modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/filter/ComplexFilterSplitter.java

```diff
@@ -131,9 +257,61 @@ public Object visit(PropertyName expression, Object notUsed) {
             postStack.push(expression);
             return null;
         }
-        
+
         List<Expression> matchingMappings = mappings.findMappingsFor(exprSteps, false);
 
+        if (AppSchemaDataAccessConfigurator.shouldEncodeNestedFilters()) {
+            // check nested mappings
+            FeatureChainedAttributeVisitor nestedAttrExtractor = new FeatureChainedAttributeVisitor(
+                    mappings);
+            nestedAttrExtractor.visit(expression, null);
+            List<FeatureChainedAttributeDescriptor> attributes = nestedAttrExtractor
+                    .getFeatureChainedAttributes();
+            // encoding of filters on multiple nested attributes is not (yet) supported
+            if (attributes.size() == 1) {
+                FeatureChainedAttributeDescriptor nestedAttrDescr = attributes.get(0);
+                if (nestedAttrDescr.chainSize() > 1 && nestedAttrDescr.isJoiningEnabled()) {
+                    nestedAttributes++;
```

Comment:
Ha ha @riniangreani, I had the same confusion, until I found this line! This is the spot where the ```nestedAttributes``` counter gets incremented, and the only write to this field that I could find that does not set it to zero.  :-)